plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -26,21 +26,27 @@
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.lang.Integer.parseInt;
 import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
 
 public final class PartitionFields
 {
-    private static final String NAME = "[a-z_][a-z0-9_]*";
-    private static final String FUNCTION_ARGUMENT_NAME = "\\((" + NAME + ")\\)";
-    private static final String FUNCTION_ARGUMENT_NAME_AND_INT = "\\((" + NAME + "), *(\\d+)\\)";
-
-    private static final Pattern IDENTITY_PATTERN = Pattern.compile(NAME);
-    private static final Pattern YEAR_PATTERN = Pattern.compile("year" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern MONTH_PATTERN = Pattern.compile("month" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern DAY_PATTERN = Pattern.compile("day" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern HOUR_PATTERN = Pattern.compile("hour" + FUNCTION_ARGUMENT_NAME);
-    private static final Pattern BUCKET_PATTERN = Pattern.compile("bucket" + FUNCTION_ARGUMENT_NAME_AND_INT);
-    private static final Pattern TRUNCATE_PATTERN = Pattern.compile("truncate" + FUNCTION_ARGUMENT_NAME_AND_INT);
-    private static final Pattern VOID_PATTERN = Pattern.compile("void" + FUNCTION_ARGUMENT_NAME);
+    private static final String UNQUOTED_IDENTIFIER = "[a-zA-Z_][a-zA-Z0-9_]*";
+    // We only support lowercase quoted identifiers for now.
+    // See https://github.com/trinodb/trino/issues/12226#issuecomment-1128839259
+    private static final String QUOTED_IDENTIFIER = "\"(?:\"\"|[^\"A-Z])*\"";
+    private static final String IDENTIFIER = "\\s*(" + UNQUOTED_IDENTIFIER + "|" + QUOTED_IDENTIFIER + ")\\s*";
+    private static final String FUNCTION_ARGUMENT_NAME = "\\(" + IDENTIFIER + "\\)\\s*";
+    private static final String FUNCTION_ARGUMENT_NAME_AND_INT = "\\(" + IDENTIFIER + ",\\s*(\\d+)\\s*\\)\\s*";
+
+    private static final Pattern UNQUOTED_IDENTIFIER_PATTERN = Pattern.compile(UNQUOTED_IDENTIFIER);
+    private static final Pattern IDENTITY_PATTERN = Pattern.compile(IDENTIFIER);
+    private static final Pattern YEAR_PATTERN = Pattern.compile("\\s*(?i)year(?-i)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern MONTH_PATTERN = Pattern.compile("\\s*(?i)month(?-i)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern DAY_PATTERN = Pattern.compile("\\s*(?i)day(?-i)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern HOUR_PATTERN = Pattern.compile("\\s*(?i)hour(?-i)" + FUNCTION_ARGUMENT_NAME);
+    private static final Pattern BUCKET_PATTERN = Pattern.compile("\\s*(?i)bucket(?-i)" + FUNCTION_ARGUMENT_NAME_AND_INT);
+    private static final Pattern TRUNCATE_PATTERN = Pattern.compile("\\s*(?i)truncate(?-i)" + FUNCTION_ARGUMENT_NAME_AND_INT);
+    private static final Pattern VOID_PATTERN = Pattern.compile("\\s*(?i)void(?-i)" + FUNCTION_ARGUMENT_NAME);
```

Comment:
Why match transform names case insensitively?
This looks like orthogonal change. In fact, it's not -- it's related to the fact you removed lowercasing in IcebergTableProperties, but the change history will not make it visible.

- let's extract prep refactor commit that removes lowercasing from IcebergTableProperties, but doesn't change connector behavior.
- for scoped case-insensitity in the pattern use `(?i:void)` instead of `(?i)void(?-i)`